src/adic_space.lean

```diff
@@ -380,29 +475,123 @@ to_stalk.rec (presheaf_of_rings.restrict U G) u (stalk_of_rings G u)
   is_ring_hom (presheaf_of_rings.restrict_stalk_map G u) :=
 by unfold presheaf_of_rings.restrict_stalk_map; apply_instance
 
-def presheaf_of_topological_rings.restrict {X : Type*} [topological_space X] (U : opens X)
+def presheaf_of_topological_rings.restrict {X : Type u} [topological_space X] (U : opens X)
   (G : presheaf_of_topological_rings X) : presheaf_of_topological_rings U :=
   { Ftop := λ V, G.Ftop (topological_space.opens.map U V),
     Ftop_ring := λ V, G.Ftop_ring (topological_space.opens.map U V),
     res_continuous := λ V W HWV, G.res_continuous (topological_space.opens.map U V)
       (topological_space.opens.map U W) (topological_space.opens.map_mono HWV),
   ..presheaf_of_rings.restrict U G.to_presheaf_of_rings }
 
-noncomputable def 𝒞.restrict {X : Type*} [topological_space X] (U : opens X) (G : 𝒞 X) : 𝒞 U :=
+noncomputable def 𝒞.restrict {X : Type u} [topological_space X] (U : opens X) (G : 𝒞 X) : 𝒞 U :=
 { F := presheaf_of_topological_rings.restrict U G.F,
   valuation :=
     λ u, Spv.mk (valuation.comap (G.valuation u).out (presheaf_of_rings.restrict_stalk_map _ _)) }
 
-structure adic_space (X : Type u) [topological_space X] :=
+noncomputable instance PreValuedRingedSpace.restrict {X : PreValuedRingedSpace.{u}} :
+  has_coe (opens X) PreValuedRingedSpace :=
+{ coe := λ U,
+  { space := U,
+    top := by apply_instance,
+    presheaf := presheaf_of_topological_rings.restrict U X.presheaf,
+    valuation :=
+      λ u, Spv.mk (valuation.comap (X.valuation u).out (presheaf_of_rings.restrict_stalk_map _ _)) } }
+
+section 𝒱
+local attribute [instance] sheaf_of_topological_rings.uniform_space
+
+/-- Wedhorn's category 𝒱 -/
+structure 𝒱 (X : Type u) [topological_space X] :=
+(ℱ : sheaf_of_topological_rings X)
+(complete : ∀ U : opens X, complete_space (ℱ.F.F U))
+(valuation : ∀ x : X, Spv (stalk_of_rings ℱ.to_presheaf_of_topological_rings.to_presheaf_of_rings x))
+(local_stalks : ∀ x : X, is_local_ring (stalk_of_rings ℱ.to_presheaf_of_rings x))
+(supp_maximal : ∀ x : X, ideal.is_maximal (_root_.valuation.supp (valuation x).out))
+
+end 𝒱
+
+def 𝒱.to_𝒞 {X : Type u} [topological_space X] (ℱ : 𝒱 X) : 𝒞 X :=
+{ F := ℱ.ℱ.to_presheaf_of_topological_rings,
+  valuation := ℱ.valuation}
+
+structure adic_space (X : Type) [topological_space X] :=
 (locally_ringed_valued_space : 𝒱 X)
 (Hlocally_affinoid : ∃ (I : Type u) (U : I → opens X) (Hcover : set.Union (λ i, (U i).1) = set.univ)
   (R : I → Huber_pair),
   ∀ i : I, nonempty (𝒞.equiv (𝒞.Spa (R i)) (𝒞.restrict (U i) locally_ringed_valued_space.to_𝒞)))
 
+section
+local attribute [instance] sheaf_of_topological_rings.uniform_space
+
+/--Category of topological spaces endowed with a sheaf of complete topological rings
+and (an equivalence class of) valuations on the stalks (which are required to be local rings).
+Wedhorn calls this category `𝒱`.-/
+structure CLVRS :=
+(space : Type) -- change this to (Type u) to enable universes
```

Comment:
I think computer scientists would be equally unhappy with that name though. Should we call it something long and stupid, or maybe even comp_loc_val_ringed_space or something?